src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -868,7 +868,44 @@ void MFEMSidreDataCollection::Load(const std::string& path,
     std::string suffixedPath = endsWith(path, ".root") ? path : path + ".root";
 
     IOManager reader(m_comm);
-    reader.read(m_bp_grp->getDataStore()->getRoot(), suffixedPath);
+    // Get the paths in which the global and domain groups are stored
+    // This allows for a datastore structure other than the one used when this
+    // class owns the DataStore
+    SLIC_ERROR_IF(!m_bp_index_grp || !m_bp_grp,
+                  "Blueprint pointers must not be null");
+    Group* global_grp = m_bp_index_grp->getParent()->getParent();
+    Group* domain_grp = m_bp_grp->getParent();
+    const std::string global_path = global_grp->getPath();
+    const std::string domain_path = domain_grp->getPath();
+
+    // Create a temporary group to read the data into
+    // This is done instead of creating a temp DataStore so the buffers are intact
+    Group* temp_root =
+      m_bp_grp->getDataStore()->getRoot()->createGroup("_sidre_tmp_load");
+    reader.read(temp_root, suffixedPath);
+
+    // First transfer the global group from the temp group to its correct location
+    Group* datastore_root = m_bp_grp->getDataStore()->getRoot();
+    Group* new_global_group = temp_root->getGroup(global_grp->getPathName());
+    Group* global_group_parent = datastore_root;
+    if(!global_path.empty())
+    {
+      global_group_parent = datastore_root->getGroup(global_path);
+    }
```

Comment:
blank lines separating the pieces described in the comments would help readability. Here, after line 897, after 906, etc.